cmake/EthCompilerSettings.cmake

```diff
@@ -56,6 +57,14 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 	eth_add_cxx_compiler_flag_if_supported(
 		$<$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>
 	)
+	eth_add_cxx_compiler_flag_if_supported(-Wfinal-dtor-non-final-class)
```

Comment:
b_ems:
```
-- Performing Test Wfinal-dtor-non-final-class
-- Performing Test Wfinal-dtor-non-final-class - Success
-- Performing Test Wnewline-eof
-- Performing Test Wnewline-eof - Success
-- Performing Test Wsuggest-destructor-override
-- Performing Test Wsuggest-destructor-override - Failed
-- Performing Test Wunreachable-code-break
-- Performing Test Wunreachable-code-break - Success
-- Performing Test Wduplicated-cond
-- Performing Test Wduplicated-cond - Failed
-- Performing Test Wduplicate-enum
-- Performing Test Wduplicate-enum - Success
-- Performing Test Wlogical-op
-- Performing Test Wlogical-op - Failed
-- Performing Test Wno-unknown-attributes
-- Performing Test Wno-unknown-attributes - Success
```

b_archlinux:
```
-- Performing Test Wfinal-dtor-non-final-class
-- Performing Test Wfinal-dtor-non-final-class - Failed
-- Performing Test Wnewline-eof
-- Performing Test Wnewline-eof - Failed
-- Performing Test Wsuggest-destructor-override
-- Performing Test Wsuggest-destructor-override - Failed
-- Performing Test Wunreachable-code-break
-- Performing Test Wunreachable-code-break - Failed
-- Performing Test Wduplicated-cond
-- Performing Test Wduplicated-cond - Success
-- Performing Test Wduplicate-enum
-- Performing Test Wduplicate-enum - Failed
-- Performing Test Wlogical-op
-- Performing Test Wlogical-op - Success
-- Performing Test Wno-unknown-attributes
-- Performing Test Wno-unknown-attributes - Failed
```

b_osx:
```
-- Performing Test Wfinal-dtor-non-final-class
-- Performing Test Wfinal-dtor-non-final-class - Failed
-- Performing Test Wnewline-eof
-- Performing Test Wnewline-eof - Success
-- Performing Test Wsuggest-destructor-override
-- Performing Test Wsuggest-destructor-override - Failed
-- Performing Test Wunreachable-code-break
-- Performing Test Wunreachable-code-break - Success
-- Performing Test Wduplicated-cond
-- Performing Test Wduplicated-cond - Failed
-- Performing Test Wduplicate-enum
-- Performing Test Wduplicate-enum - Success
-- Performing Test Wlogical-op
-- Performing Test Wlogical-op - Failed
-- Performing Test Wno-unknown-attributes
-- Performing Test Wno-unknown-attributes - Success
```

b_ubu_cxx20:
```
-- Performing Test Wfinal-dtor-non-final-class
-- Performing Test Wfinal-dtor-non-final-class - Failed
-- Performing Test Wnewline-eof
-- Performing Test Wnewline-eof - Failed
-- Performing Test Wsuggest-destructor-override
-- Performing Test Wsuggest-destructor-override - Failed
-- Performing Test Wunreachable-code-break
-- Performing Test Wunreachable-code-break - Failed
-- Performing Test Wduplicated-cond
-- Performing Test Wduplicated-cond - Success
-- Performing Test Wduplicate-enum
-- Performing Test Wduplicate-enum - Failed
-- Performing Test Wlogical-op
-- Performing Test Wlogical-op - Success
-- Performing Test Wno-unknown-attributes
-- Performing Test Wno-unknown-attributes - Failed
```

b_ubu:
```
-- Performing Test Wfinal-dtor-non-final-class
-- Performing Test Wfinal-dtor-non-final-class - Failed
-- Performing Test Wnewline-eof
-- Performing Test Wnewline-eof - Failed
-- Performing Test Wsuggest-destructor-override
-- Performing Test Wsuggest-destructor-override - Failed
-- Performing Test Wunreachable-code-break
-- Performing Test Wunreachable-code-break - Failed
-- Performing Test Wduplicated-cond
-- Performing Test Wduplicated-cond - Success
-- Performing Test Wduplicate-enum
-- Performing Test Wduplicate-enum - Failed
-- Performing Test Wlogical-op
-- Performing Test Wlogical-op - Success
-- Performing Test Wno-unknown-attributes
-- Performing Test Wno-unknown-attributes - Failed
```

b_ubu_clang:
```
-- Performing Test Wfinal-dtor-non-final-class
-- Performing Test Wfinal-dtor-non-final-class - Success
-- Performing Test Wnewline-eof
-- Performing Test Wnewline-eof - Success
-- Performing Test Wsuggest-destructor-override
-- Performing Test Wsuggest-destructor-override - Failed
-- Performing Test Wunreachable-code-break
-- Performing Test Wunreachable-code-break - Success
-- Performing Test Wduplicated-cond
-- Performing Test Wduplicated-cond - Failed
-- Performing Test Wduplicate-enum
-- Performing Test Wduplicate-enum - Success
-- Performing Test Wlogical-op
-- Performing Test Wlogical-op - Failed
-- Performing Test Wno-unknown-attributes
-- Performing Test Wno-unknown-attributes - Success
```

So it seems `-Wsuggest-destructor-override` is not tested on any target, should just remove it.

Also suggest which ones don't make sense for any of you, these came from https://github.com/ethereum/solidity/issues/6905#issuecomment-725480130

Reply:
b_ubu also supports:
```
-- Performing Test Wcast-align=strict
-- Performing Test Wcast-align=strict - Success
```

Reply:
Not sure ``final-dtor-non-final-class`` (which also seems clang only - not that it hurts *that* much) applies to us that much, since we don't use ``final`` anyways, which is what it's about it seems.

Reply:
As for ``suggest-destructor-override`` - I personally find marking destructors ``override`` a bit weird, because I never do, but it does have some point in that it's used to assure that the base constructor is properly virtual... so fine either way on that one...

Reply:
``-Wunreachable-code-break`` I'm also rather impartial about... e.g. in ``case Sth: solAssert(false, ""); break;`` I'm fine with and without the ``break`` myself - this would mean dropping it. I thought about whether it'd be dangerous if the assert stops failing for whatever reason, but we'll probably have fallthrough warnings in that case ;-).

Reply:
``-Wduplicated-cond`` I find rather exotic, but it definitely doesn't hurt - I don't think it'll ever catch anything, but still all for it, one never knows!

Reply:
Similar for ``-Wduplicate-enum`` - not sure our current settings would accept that, but it's more likely to happen sometime and is good to have safety against. (it's about ``enum E { A = 1, B, C = 2 }``-style conflicts)

Reply:
``-Wlogical-op`` is specified rather vaguely, so depends on what "suspicious use of a logical operator" entails. But maybe no hurt in it - we can still disable if it hits something legitimate.

Reply:
``Wno-unknown-attributes`` is also unlikely relevant for us, but doesn't hurt.

Reply:
That was the entire list, right? A few of them are clang only - that may be a bit inconvenient, but I wouldn't expect them to happen a lot anyways, so probably fine. Did you actually check that none of them are implicit in ``-Wall``? A few of them like ``duplicate-enum`` I'd have expected to be.

Reply:
> Did you actually check that none of them are implicit in -Wall

We did when we added this list to Fizzy.

Reply:
> As for suggest-destructor-override

I guess we really just have to remove this because the negation of it doesn't work and it breaks the ossfuzz build (as the protobuf generator goes against this suggestion).